db_stress_tool/cf_consistency_stress.cc

```diff
@@ -491,6 +494,50 @@ class CfConsistencyStressTest : public StressTest {
     } while (true);
   }
 
+#ifndef ROCKSDB_LITE
+  void ContinuouslyVerifyDb(ThreadState* thread) const override {
+    assert(cmp_db_ != nullptr);
+    Status s = cmp_db_->TryCatchUpWithPrimary();
+    ReadOptions ropts;
+    ropts.total_order_seek = true;
+    const auto checksum_column_family = [](Iterator* iter,
+                                           uint32_t* checksum) -> Status {
+      assert(nullptr != checksum);
+      uint32_t ret = 0;
+      for (iter->SeekToFirst(); iter->Valid(); iter->Next()) {
+        ret = crc32c::Extend(ret, iter->key().data(), iter->key().size());
+        ret = crc32c::Extend(ret, iter->value().data(), iter->value().size());
+      }
+      *checksum = ret;
+      return iter->status();
+    };
+    SharedState* shared = thread->shared;
+    Status status;
+    uint32_t crc = 0;
+    {
+      // Compute crc for all key-values of default column family.
+      std::unique_ptr<Iterator> it(cmp_db_->NewIterator(ropts));
+      status = checksum_column_family(it.get(), &crc);
+    }
+    uint32_t tmp_crc = 0;
+    if (status.ok()) {
+      for (ColumnFamilyHandle* cfh : cmp_cfhs_) {
+        if (cfh == cmp_db_->DefaultColumnFamily()) {
+          continue;
+        }
+        std::unique_ptr<Iterator> it(cmp_db_->NewIterator(ropts, cfh));
+        status = checksum_column_family(it.get(), &tmp_crc);
+        if (!status.ok() || tmp_crc != crc) {
+          break;
+        }
+      }
+    }
+    if (!status.ok() || tmp_crc != crc) {
+      shared->SetShouldStopTest();
```

Comment:
Does it make the test run fail? I didn't figure out in which mechanism the test would fail.

Reply:
Yes it does. In `OperateDb()`, we check whether `shared->ShouldStopTest()` in each iteration.

After the continuous verification thread calls `shared->SetShouldStopTest()`, the other test threads can see this change and stop the operations. Then we perform the original key-by-key comparison and report the test results.
With this being said, you can see that we have two verification logic: one in the continuous verification thread that computes checksums, the other done at the end of the test, reporting exact key that mismatches.